composer/trainer/dist_strategy.py

```diff
@@ -369,78 +369,141 @@ def sync_hook(*args):
             if hasattr(obj, '_fsdp_wrap') and not bool(obj._fsdp_wrap):
                 continue
 
-            def _param_init_fn(module: torch.nn.Module) -> None:
-                # A dictionary of all tied parameter pointers to module names
+            # Rather than verifying these changes with older PyTorch versions, we are fixing forward here
+            if version.parse(torch.__version__) > version.parse('2.1.0'):
+                # A dictionary of all tied parameter pointers to (module, attr) tuples
                 tied_pointers = {}
```

Comment:
looks like duplicate code. can weight_tying logic be made into a function?

Reply:
its actually not duplicate code. I'm saving a different thing in the dict for this version. It may be possible to unify to the new implementation, but I wanted to avoid touching the previous code.